freqtrade/rpc/api_server.py

```diff
@@ -0,0 +1,339 @@
+            pass
+        else:
+            return list(iterable)
+        return JSONEncoder.default(self, obj)
+
+
+class ApiServer(RPC):
+    """
+    This class runs api server and provides rpc.rpc functionality to it
+
+    This class starts a none blocking thread the api server runs within
+    """
+
+    def safe_rpc(func):
+
+        def func_wrapper(self, *args, **kwargs):
+
+            try:
+                return func(self, *args, **kwargs)
+            except RPCException as e:
+                logger.exception("API Error calling %s: %s", func.__name__, e)
+                return self.rest_error(f"Error querying {func.__name__}: {e}")
+
+        return func_wrapper
+
+    def __init__(self, freqtrade) -> None:
+        """
+        Init the api server, and init the super class RPC
+        :param freqtrade: Instance of a freqtrade bot
+        :return: None
+        """
+        super().__init__(freqtrade)
+
+        self._config = freqtrade.config
+        self.app = Flask(__name__)
+        self.app.json_encoder = ArrowJSONEncoder
+
+        # Register application handling
+        self.register_rest_other()
+        self.register_rest_rpc_urls()
+
+        thread = threading.Thread(target=self.run, daemon=True)
+        thread.start()
+
+    def cleanup(self) -> None:
+        logger.info("Stopping API Server")
+        self.srv.shutdown()
+
+    def run(self):
+        """
+        Method that runs flask app in its own thread forever.
+        Section to handle configuration and running of the Rest server
+        also to check and warn if not bound to a loopback, warn on security risk.
+        """
+        rest_ip = self._config['api_server']['listen_ip_address']
+        rest_port = self._config['api_server']['listen_port']
+
+        logger.info(f'Starting HTTP Server at {rest_ip}:{rest_port}')
+        if not IPv4Address(rest_ip).is_loopback:
+            logger.warning("SECURITY WARNING - Local Rest Server listening to external connections")
+            logger.warning("SECURITY WARNING - This is insecure please set to your loopback,"
+                           "e.g 127.0.0.1 in config.json")
+
+        # Run the Server
+        logger.info('Starting Local Rest Server')
+        try:
+            self.srv = make_server(rest_ip, rest_port, self.app)
+            self.srv.serve_forever()
+        except Exception:
+            logger.exception("Api server failed to start, exception message is:")
+        logger.info('Starting Local Rest Server_end')
+
+    def send_msg(self, msg: Dict[str, str]) -> None:
+        """
+        We don't push to endpoints at the moment.
+        Take a look at webhooks for that functionality.
+        """
+        pass
+
+    def rest_dump(self, return_value):
+        """ Helper function to jsonify object for a webserver """
+        return jsonify(return_value)
+
+    def rest_error(self, error_msg):
+        return jsonify({"error": error_msg}), 502
+
+    def register_rest_other(self):
```

Comment:
I find `register_rest_other` confusing. probably `register_rest_base` ? or something similar ?

Reply:
it doesn't make sense - it used to have more endpoints ... but most were removed / replaced ... so i'll remove this method.